samples/csharp-language/delegates-and-events/FileLogger.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.IO;
+
+namespace DelegatesAndEvents
+{
+public class FileLogger
+{
+    private readonly string logPath;
+    public FileLogger(string path)
+    {
+        logPath = path;
+        Logger.WriteMessage += LogMessage;
+    }
+    
+    public void DetachLog() => Logger.WriteMessage -= LogMessage;
+    // make sure this can't throw.
+    private void LogMessage(string msg)
+    {
+        try {
+            using (var log = File.AppendText(logPath))
+            {
+                log.WriteLine(msg);
+                log.Flush();
+            }
+        } catch (Exception e)
+        {
+            // Hmm. Not sure what to do.
+            // Logging is failing...
```

Comment:
Good point. I updated the comment to explain why there is a catch clause that does nothing.